kratos/containers/variables_list_data_value_container.h

```diff
@@ -325,32 +325,34 @@ class KRATOS_API(KRATOS_CORE) VariablesListDataValueContainer
     template<class TDataType>
     TDataType& GetValue(const Variable<TDataType>& rThisVariable)
     {
-        if(!mpVariablesList->Has(rThisVariable))
-            KRATOS_THROW_ERROR(std::invalid_argument, "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:",rThisVariable);
+        KRATOS_ERROR_IF_NOT(mpVariablesList->Has(rThisVariable)) << "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:" << rThisVariable << std::endl;
         return *(TDataType*)Position(rThisVariable);
     }
 
     template<class TDataType>
     TDataType& GetValue(const Variable<TDataType>& rThisVariable, SizeType QueueIndex)
     {
-        if(!mpVariablesList->Has(rThisVariable))
-            KRATOS_THROW_ERROR(std::invalid_argument, "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:",rThisVariable);
+        KRATOS_ERROR_IF_NOT(mpVariablesList->Has(rThisVariable)) << "This container only can store the variables specified in its variables list. The variables list doesn't have this variable:" << rThisVariable << std::endl;
+    #ifdef KRATOS_DEBUG       
+        KRATOS_ERROR_IF((QueueIndex + 1) > mQueueSize) << "ERROR:: YOUR ARE TRYING TO ACCES A VALUE OUTSIDE THE BUFFER SIZE: " << mQueueSize << std::endl;
```

Comment:
There is a `KRATOS_DEBUG_ERROR_IF` that fits here. It would be good to use it, in case we ever decide to change the debug condition.

Reply:
I didn't know